quit/basicfunction.py

```diff
@@ -0,0 +1,121 @@
+import typing
+from collections.abc import Iterable
+
+import numpy as np
+from scipy import linalg
+
+
+def ket(psi: Iterable[complex]) -> np.ndarray:
+    """Create a ket |psi> as a vertical vector.
+
+    :param ket: Sequence of complex numbers.
+    :return: A vertical vector as numpy array.
+    """
+    return np.reshape(np.asarray(psi), (-1, 1))
```

Comment:
It seems that this function would succeed also for non 1-D inputs, returning somewhat surprising results. What do you think about checking that the input is indeed "sequence-like", that is:

- casting it to array using `np.asarray`
- checking if after using `squeeze` the resulting array has only a single dimension

?